content/guides/getstarted/rust-to-solana.md

```diff
@@ -0,0 +1,186 @@
+
+```rust 
+msg!(“Your message”);
+msg!(0_64, 1_64, 2_64);
+msg!(“Your variable: {:?}”, variable);
+```
+
+The `panic!`, `assert!`, and any internal panics are also output to the program logs by default. However, this can be modified with a custom panic handler.
+
+## Compute Budget 
+As a Rust developer, efficient computing is nothing new. What may be different is that in Solana, each transaction has a fixed [compute budget](https://solana.com/docs/core/runtime#compute-budget) that it must not surpass. When transactions exceed the compute budget, they are halted and return an error.
+
+Programs can access the number of remaining compute units via the `sol_remaining_compute_units` system call, and can log the remaining number of compute units with `sol_log_compute_units`.
+
+## Learning the Development Environment and Frameworks
+While the Solana CLI and the `solana_program` crate are all that is needed to get started, there are a couple of helpful tools which can accelerate learning.
+
+<div align="center">
+    <img src="./rust-to-solana-img/network-diagram.png" alt="Network Diagram" style="max-width: 80%; height: auto;"/>
+</div>
+
+### Solana Playground 
+The [Solana Playground](https://beta.solpg.io/) is a browser-based IDE that allows developers to develop and deploy Solana programs.
+
+<div align="center">
+    <img src="./rust-to-solana-img/solana-playground.png" alt="Solana Playground" style="max-width: 80%; height: auto;"/>
+</div>
+
+It’s the easiest way to begin developing with Solana, and it supports building, testing, and deploying Solana Rust programs. Additionally, a number of built-in tutorials are available to guide learning.
+
+The Solana Playground allows developers to get started immediately without setting up a local environment.
+
+### Using Anchor 
+[Anchor](https://www.anchor-lang.com/) is a framework that seeks to accelerate the building of secure Solana programs. It can help by handling standard boilerplate code, speeding up the development cycle. Additionally, it provides some security checks by default, making Solana programs more secure.
+
+To create a new program, simply create a new Anchor project in the Solana playground. 
+
+Alternatively, install the Anchor CLI locally, and then use `anchor init <project-name>` to create a new Anchor project.
+
+## Creating Off-chain Programs 
+So far, this guide has covered the key details of developing **on-chain Solana programs** in Rust. However, it’s also possible to develop **off-chain Solana clients** in Rust. This can be done by using the [solana_sdk crate](https://docs.rs/solana-sdk/latest/solana_sdk/). This contains the [solana_client crate](https://docs.rs/solana-client/latest/solana_client/) that allows Rust programs to interact with a Solana node via the JSON RPC API.
```

Comment:
Add link to JSON RPC API docs?

Reply:
Would rust devs know here how to add the solana crate? I would not :D Is it adding the solana crate the to the cargo.toml i guess?

Reply:
yup adding crates is needed for all rust dev stuff :)